src/lib/types.ts

```diff
@@ -1131,3 +1191,224 @@ export type SignOut = {
+  clientDataJSON: Base64URLString
+  authenticatorData: Base64URLString
+  signature: Base64URLString
+  userHandle?: Base64URLString
+}
+
+/**
+ * A WebAuthn-compatible device and the information needed to verify assertions by it
+ */
+export type AuthenticatorDevice = {
+  credentialID: Base64URLString
+  credentialPublicKey: Uint8Array
+  // Number of times this authenticator is expected to have been used
+  counter: number
+  // From browser's `startRegistration()` -> RegistrationCredentialJSON.transports (API L2 and up)
+  transports?: AuthenticatorTransportFuture[]
+}
+
+/**
+ * An attempt to communicate that this isn't just any string, but a Base64URL-encoded string
+ */
+export type Base64URLString = string
+
+/**
+ * AuthenticatorAttestationResponse in TypeScript's DOM lib is outdated (up through v3.9.7).
+ * Maintain an augmented version here so we can implement additional properties as the WebAuthn
+ * spec evolves.
+ *
+ * See https://www.w3.org/TR/webauthn-2/#iface-authenticatorattestationresponse
+ *
+ * Properties marked optional are not supported in all browsers.
+ */
+export interface AuthenticatorAttestationResponseFuture extends AuthenticatorAttestationResponse {
+  getTransports(): AuthenticatorTransportFuture[]
+}
+
+/**
+ * A super class of TypeScript's `AuthenticatorTransport` that includes support for the latest
+ * transports. Should eventually be replaced by TypeScript's when TypeScript gets updated to
+ * know about it (sometime after 4.6.3)
+ */
+export type AuthenticatorTransportFuture =
+  | 'ble'
+  | 'cable'
+  | 'hybrid'
+  | 'internal'
+  | 'nfc'
+  | 'smart-card'
+  | 'usb'
+
+/**
+ * A super class of TypeScript's `PublicKeyCredentialDescriptor` that knows about the latest
+ * transports. Should eventually be replaced by TypeScript's when TypeScript gets updated to
+ * know about it (sometime after 4.6.3)
+ */
+export interface PublicKeyCredentialDescriptorFuture
+  extends Omit<PublicKeyCredentialDescriptor, 'transports'> {
+  transports?: AuthenticatorTransportFuture[]
+}
+
+/** */
+export type PublicKeyCredentialJSON = RegistrationResponseJSON | AuthenticationResponseJSON
+
+/**
+ * A super class of TypeScript's `PublicKeyCredential` that knows about upcoming WebAuthn features
+ */
+export interface PublicKeyCredentialFuture extends PublicKeyCredential {
+  type: PublicKeyCredentialType
+  // See https://github.com/w3c/webauthn/issues/1745
+  isConditionalMediationAvailable?(): Promise<boolean>
+  // See https://w3c.github.io/webauthn/#sctn-parseCreationOptionsFromJSON
+  parseCreationOptionsFromJSON?(
+    options: PublicKeyCredentialCreationOptionsJSON
+  ): PublicKeyCredentialCreationOptions
+  // See https://w3c.github.io/webauthn/#sctn-parseRequestOptionsFromJSON
+  parseRequestOptionsFromJSON?(
+    options: PublicKeyCredentialRequestOptionsJSON
+  ): PublicKeyCredentialRequestOptions
+  // See https://w3c.github.io/webauthn/#dom-publickeycredential-tojson
+  toJSON?(): PublicKeyCredentialJSON
+}
```

Comment:
There's just so much going on here, and doesn't feel very used.

Can we just simplify this... the TypeScript compiler already includes all of the types necessary. We don't need to redefine anything.

Reply:
The entire section is done as sugar around the TypeScript compiler fields. In the TypeScript version, certain fields are [BufferSource](https://github.com/MasterKale/SimpleWebAuthn/blob/5e3e5718f6b97ee3df09468a4400d3c7770a3f31/packages/types/src/dom.ts#L95) while in the sugar version suffixed with `-JSON` these fields are cast to `base64URL / string` which is slightly easier to work with.

I would also like to remove this if possible, will see if there's a clean workaround. In the meantime open to suggestions as well